src/Oscoin/Crypto/Hash.hs

```diff
@@ -108,10 +112,16 @@ class
     -- | The zero hash. Also the minimum hash value.
     zeroHash  :: Hash crypto
 
+    -- | The zero short hash. Also the minimum short hash value.
+    zeroShortHash  :: ShortHash crypto
+
     -- | A short hash, in the similar spirit of the BTC one.
     -- eg. ripemd160(blake2b(x))
     toShortHash :: Hash crypto -> ShortHash crypto
 
+    -- | Parse a 'ShortHash' from 'Text'.
+    parseShortHash :: Text -> Maybe (ShortHash crypto)
```

Comment:
It's slightly strange that we have this `parseShortHash` but not the equivalent for the "normal" one. I do agree though that this interface grew up a little bit out of control. For now let's stick with what you have done here and perhaps in the next iteration we will think at how we can make the API a bit more consistent.

Reply:
Yeah - I didn't want to add `parseHash` if we weren't actually using it, but I agree that we should look at this at some point (and actually the whole module).